src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -603,7 +608,7 @@ private Settings buildIndexSettings(IndexMetaData metaData) {
     }
 
     /**
-     * Adds a pending delete for the given index.
+     * Adds a pending delete for the given index shard.
      */
     public void addPendingDelete(Index index, ShardId shardId, Settings settings) {
```

Comment:
can we mark shardId as nullable and document it means the deletion of the entire index? o.w. maybe it's better to this method not accept null and have a private variant and have the null shardId as a a whole index delete flag be an implementation detail.